main.go

```diff
@@ -1,22 +1,47 @@
 package main
 
 import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
 	"net/http"
 	"os"
 )
 
 func indexHandler(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("<h1>Hello Liatrio!</h1>"))
+	_, err := w.Write([]byte("<h1>Hello Liatrio!</h1>"))
+
+	log.Debugf("Request received on %s", r.RequestURI)
+
+	if err != nil {
+		log.Errorf("error sending response: %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	hellosGiven.Inc()
 }
 
+var (
+	hellosGiven = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
This basically becomes a dupe of `promhttp_metric_handler_requests_total{code="200"}`, but I thought it would be interesting to have some custom metrics